src/main/java/com/_up/megastore/services/implementations/UserService.java

```diff
@@ -115,19 +119,30 @@ private void ifUsernameAlreadyExistsThrowException(String username) {
   }
 
   private User findUserToActivateOrThrowException(UUID userId,
-      UUID activationToken) {
-    return userRepository.findByUserIdAndActivationTokenAndActivatedIsFalse(userId,
-            activationToken)
+      UUID activationTokenUUID) {
+    return userRepository.findByUserIdAndActivatedIsFalse(userId).filter(user -> user.getActivationTokens().stream().anyMatch( activationToken-> activationToken.getActivationToken().equals(activationTokenUUID)))
         .orElseThrow(
             () -> new IllegalArgumentException("Token is invalid or user is already activated"));
   }
 
-  private void generateActivationToken(User user){
-    user.setActivationToken(UUID.randomUUID());
-    user.setTokenExpirationDate(LocalDateTime.now().plusHours(24));
+  private UUID generateActivationToken(User user){
+    Token activationToken = new Token();
+    UUID activationTokenUUID = randomUUID();
+    activationToken.setActivationToken(activationTokenUUID);
+    activationToken.setUser(user);
+    List<Token> activationTokens = user.getActivationTokens();
+    activationTokens.add(activationToken);
+    user.setActivationTokens(activationTokens);
```

Comment:
No estoy seguro sobre esto pero creo que se podría simplificar
```suggestion
    user.getActivationTokens().add(activationToken);
```
Tengo entendido que, como la lista es la misma instancia mutable que está asociada al objeto user, debería actualizarse sólo con el add